backend/rate_limiter.py

```
@@ -0,0 +1,313 @@
+
+        Args:
+            attempt: The retry attempt number (0-indexed)
+
+        Returns:
+            Delay in seconds
+        """
+        delay = self.base_delay * (2 ** attempt)
+
+        if self.jitter:
+            # Add random jitter: ±10% of delay
+            jitter_amount = delay * 0.1 * random.random()
```

Ruff (0.15.1)

[error] Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)